client/app/home/home.controller.js

```diff
@@ -25,6 +28,43 @@ class HomeCtrl {
         return this.guides.sections;
       });
   }
+
+  onNewRegionsAvailableMessageDismiss() {
+    this.dismissNewRegionsAvailableKey(this.newRegionsAvailableMessageKey);
+  }
+
+  initNewRegionsAvailableMessage(key) {
+    this.getNewRegionsAvailableKey(key)
```

Comment:
`getNewRegionsAvailableKey` returns a `Promise`, so you should return its return value

Reply:
not required. if the key is not found, show the message on UI and create it in the background, no need to wait for its status.

Reply:
`return`ing `Promise`s sends a message that this function can be chained, it doesn't have to have a real meaning